content/source/docs/enterprise/system-overview/architecture.html.md

```diff
@@ -5,24 +5,201 @@ page_title: "Architecture Summary - System Overview - Terraform Enterprise"
+
+### Background Services
+
+This is the largest part of the Application Layer and includes multi-purpose Sidekiq workers, Go services that provide key functionality of Terraform Enterprise, and a dynamic fleet of isolated execution environments that perform Terraform runs.
+
+## Coordination Layer
+
+The Coordination Layer is responsible for queuing asynchronous jobs and distributing them to background services running in the Application Layer. Standalone deployments will start a local Redis container for this purpose and Active/Active deployments will use an external Redis service.
+
+![tfe-int-coordination-layer](assets/tfe-int-coordination-layer.png)
+
+~> **Note:** While these components are critical to the overall operation of Terraform Enterprise, operators rarely (if ever) need to look at them directly.
+
+## Storage Layer
+
+All application data is stored in PostgreSQL and object storage. If you have selected Mounted Disk mode, local services will be started for you, otherwise, the appropriate services will be configured to access a external PostgreSQL, S3-compatible blob storage, and, in active-active, Redis.
+
+~> **Note:** The configuration information provided and/or generated during installation (e.g. database credentials, hostname, etc.) is stored outside of the application.
+
+## Key Component Interactions
+
+This section details how the key containers work together to accomplish common tasks.
+
+### Initial Startup - External Services Mode
+
+![tfe-int-start-external-services](assets/tfe-int-start-external-services.png)
+
+1. The first batch of services starts in parallel.
+2. Once `ptfe_nginx` has started, `rabbitmq` starts.
+
+### Initial Startup - Mounted Disk Mode
+
+![tfe-int-start-mounted-disk](assets/tfe-int-start-mounted-disk.png)
+
+1. The first batch of services starts in parallel.
+2. Once `ptfe_nginx` has started, `rabbitmq` starts.
+3. Once `rabbitmq` has started, `postgres` starts.
+
+### Wait for Dependencies
+
+The application waits for `postgres`, `redis`, and `rabbitmq` to start as above. Once these key services respond successfully to requests, the startup process completes with the following steps.
+
+![tfe-int-wait-for-dependencies](assets/tfe-int-wait-for-dependencies.png)
+
+1. The remaining services that do not rely on PostgreSQL are launched in parallel. Included in this batch are or two short lived containers, `ptfe_migrations` and `ptfe_registry_migrations`, which exist only to apply any necessary updates to the database schema.<br />
+2. The main **web** and **background** services start up:
+<br />2a. Once `ptfe_migrations` has finished, `ptfe_atlas` and `ptfe_sidekiq` are started in parallel.
+<br />2b. Once `ptfe_registry_migrations` has finished, `ptfe_registry_api` and `ptfe_registry_worker` are started in parallel.
+
+Finally, startup is complete and the application is ready to go at your configured hostname.
```

Comment:
I believe we should remove this section entirely as this is subject to change and actually changing as we speak.

Reply:
What is it changing to and when will the release happen? I'll comment `Wait for Dependencies`.